### PR TITLE
Added task to print coverage to logs

### LINKED ISSUE #229: Print total coverage in console
After the `koverMergedHtmlReport` task is executed, I only see the link to the report, so to check the actual total coverage value I need to open the report itself. So I want to avoid that. 

Is there any way to print the total coverage value in the console?

### PR DESCRIPTION
**Pull request by shanshin**
Resolves #229

### CODE REVIEW

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt [hunk 1]
```diff
@@ -221,6 +221,91 @@ public interface KoverReportsConfig {
...
+
+    /**
+     * Add a header line to be output before the lines with coverage.
+     *
+     * By default, the header is missing.
+     */
+    public var header: String?
+
+    /**
+     * Format of the string to output coverage for the specified in [groupBy] group.
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt:282]:
```suggestion
     * Format of the strings to print coverage for the specified in [groupBy] group.
```

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt [hunk 2]
```diff
@@ -221,6 +221,91 @@ public interface KoverReportsConfig {
...
+     *
+     * Placeholders used:
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt:284]:
```suggestion
     * The following placeholders can be used:
```

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt [hunk 3]
```diff
@@ -221,6 +221,91 @@ public interface KoverReportsConfig {
...
+     *  - `<value>` - coverage value
+     *  - `<entity>` - name of the entity by which the grouping took place. `application` if [groupBy] is [GroupingEntityType.APPLICATION].
+     *
+     * Default value is `<entity> line coverage: <value>%`.
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt:288]: Documentation doesn't mention what happens if I set `format = null`? Default value is used?

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt [hunk 4]
```diff
@@ -221,6 +221,91 @@ public interface KoverReportsConfig {
...
+    public var groupBy: GroupingEntityType?
+
+    /**
+     * Specifies which metric is used for code coverage verification.
+     * Default is [MetricType.LINE]
+     */
+    public var coverageUnits: MetricType?
+
+    /**
+     * Specifies type of lines counter value to compare with minimal and maximal values if them defined.
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt:305]: Is this value really 'compared'? Or just 'printed'?
Also, not clear what 'if them defined' means. 'If they are defined'?

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tasks/reports/KoverFormatCoverageTask.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    abstract val groupBy: Property<GroupingEntityType>
+
+    @get:Input
+    abstract val coverageUnits: Property<MetricType>
+
+    @get:Input
+    abstract val aggregationForGroup: Property<AggregationType>
+
+    @get:OutputFile
+    abstract val outputFile: RegularFileProperty
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tasks/reports/KoverFormatCoverageTask.kt:36]: Should this be configurable via `KoverLogReportConfig` or is it some internal setting? Is this file supposed to be seen by user?

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/jacoco/Evaluation.kt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import java.math.BigDecimal
+
+internal fun ReportContext.printJacocoCoverage(request: CoverageRequest, outputFile: File) {
+    val bound = VerificationBound(ONE_HUNDRED, BigDecimal.ZERO, request.metric, request.aggregation)
+    val failRule = VerificationRule(true, null, null, request.entity, listOf(bound))
+
+    val violations = doJacocoVerify(listOf(failRule))
+    val values = violations.flatMap { rule ->
+        if (rule.bounds.isEmpty()) {
+            throw KoverCriticalException("Expected only one bound violation for JaCoCo")
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/jacoco/Evaluation.kt:26]:
```suggestion
            throw KoverCriticalException("Expected at least one bound violation for JaCoCo")
```

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/kover/Evaluation.kt [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+
+internal abstract class CollectCoverageAction : WorkAction<CollectCoverageParameters> {
+    override fun execute() {
+        val request = parameters.request.get()
+        val bound = VerificationBound(ONE_HUNDRED, BigDecimal.ZERO, request.metric, request.aggregation)
+        val failRule = VerificationRule(true, null, null, request.entity, listOf(bound))
+
+        // dirty hack, add this feature to reporter
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/kover/Evaluation.kt:47]: Should a corresponding issue be created?

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/jacoco/Evaluation.kt [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import kotlinx.kover.gradle.plugin.tools.writeToFile
+import kotlinx.kover.gradle.plugin.util.ONE_HUNDRED
+import java.io.File
+import java.math.BigDecimal
+
+internal fun ReportContext.printJacocoCoverage(request: CoverageRequest, outputFile: File) {
+    val bound = VerificationBound(ONE_HUNDRED, BigDecimal.ZERO, request.metric, request.aggregation)
+    val failRule = VerificationRule(true, null, null, request.entity, listOf(bound))
+
+    val violations = doJacocoVerify(listOf(failRule))
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/jacoco/Evaluation.kt:23]: Shouldn't we also check that `violations` contain only one violation, as in Kover tool part?

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/kover/Evaluation.kt [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        val violations = koverVerify(
+            listOf(failRule),
+            parameters.filters.get(),
+            parameters.tempDir.get().asFile,
+            parameters.files.get()
+        )
+
+        val violation = violations.singleOrNull() ?: throw KoverCriticalException("Expected only one rule violation for Kover")
+        if (violation.bounds.isEmpty()) {
+            throw KoverCriticalException("Expected only one bound violation for Kover")
```
Comment (sandwwraith) [inline kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/kover/Evaluation.kt:57]:
```suggestion
            throw KoverCriticalException("Expected at least one bound violation for Kover")
```

**Review by sandwwraith — CHANGES_REQUESTED** [review]

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tasks/reports/KoverFormatCoverageTask.kt (hunk 5 above)
Comment (shanshin) [reply kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tasks/reports/KoverFormatCoverageTask.kt:36]: These are implementation details, this task is intended only for printing to the log. With the appropriate request, we can add it later

File: kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/jacoco/Evaluation.kt (hunk 8 above)
Comment (shanshin) [reply kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/tools/jacoco/Evaluation.kt:23]: No, due to the implementation details, an list of rule violations containing only one bound violation is returned for JaCoCo.

**Review by sandwwraith — APPROVED** [review]

**Pull request merged**